File: internal/model/dto/response/sync.go
package response

import "time"

// SyncSourceConfigResponse 描述同步源非敏感配置响应
type SyncSourceConfigResponse struct {
	OperatorUnionID string `json:"operator_union_id"`
	WorkspaceID     string `json:"workspace_id"`
	RootNodeID      string `json:"root_node_id"`
	SyncMode        string `json:"sync_mode"`
}

// SyncSourceResponse 描述同步源响应
type SyncSourceResponse struct {
	ID               string                   `json:"id"`
	KnowledgeBaseID  string                   `json:"knowledge_base_id"`
	Name             string                   `json:"name"`
	Platform         string                   `json:"platform"`
	SourceConfig     SyncSourceConfigResponse `json:"source_config"`
	Status           int                      `json:"status"`
	LastSyncAt       *time.Time               `json:"last_sync_at"`
	LastErrorMessage string                   `json:"last_error_message"`
	CreatedAt        time.Time                `json:"created_at"`
	UpdatedAt        time.Time                `json:"updated_at"`
	DeletedAt        *time.Time               `json:"deleted_at"`
}

// SyncJobResponse 描述同步任务响应
type SyncJobResponse struct {
	ID              string     `json:"id"`
	SyncSourceID    string     `json:"sync_source_id"`
	KnowledgeBaseID string     `json:"knowledge_base_id"`
	JobType         string     `json:"job_type"`
	Status          int        `json:"status"`
	TotalCount      int        `json:"total_count"`
	SuccessCount    int        `json:"success_count"`
	FailedCount     int        `json:"failed_count"`
	ErrorMessage    string     `json:"error_message"`
	StartedAt       *time.Time `json:"started_at"`
	FinishedAt      *time.Time `json:"finished_at"`
	CreatedAt       time.Time  `json:"created_at"`
	UpdatedAt       time.Time  `json:"updated_at"`
}

File: design/vue/src/types/sync.ts
export interface SyncSourceConfig {
  operator_union_id?: string
  workspace_id: string
  root_node_id: string
  sync_mode: string
}

export interface SyncSource {
  id: string
  knowledge_base_id: string
  name: string
  platform: string
  source_config: SyncSourceConfig
  status: number
  last_sync_at: string | null
  last_error_message: string
  created_at: string
  updated_at: string
  deleted_at: string | null
}

export interface CreateSyncSourceRequest {
  knowledge_base_id: string
  name: string
  platform: 'dingtalk'
  source_config: SyncSourceConfig
}

export interface UpdateSyncSourceRequest {
  name: string
  source_config: SyncSourceConfig
  status?: number
}

export interface SyncJob {
  id: string
  sync_source_id: string
  knowledge_base_id: string
  job_type: string
  status: number
  total_count: number
  success_count: number
  failed_count: number
  error_message: string
  started_at: string | null
  finished_at: string | null
  created_at: string
  updated_at: string
}

export interface SyncItem {
  id: string
  sync_source_id: string
  knowledge_base_id: string
  external_id: string
  parent_external_id: string
  name: string
  item_type: string
  category: string
  extension: string
  external_url: string
  file_size: number
  has_children: boolean
  source_updated_at: string | null
  local_document_id: string
  import_status: number
  error_message: string
  created_at: string
  updated_at: string
}
